pkg/vtpm/src/server.cpp

```diff
@@ -0,0 +1,173 @@
+            return -1;
+        }
+        input_file << file.content();
+        input_file.close();
+    }
+
+    ostringstream command;
+    command << request.command() << " " << "> cmd.output 2>&1";
+    system(command.str().c_str());
+    eve_tools::EveTPMResponse response;
+    ifstream cmdOut;
+    cmdOut.open("cmd.output", ios::in);
+    if (!cmdOut) {
+        cout << "Error opening cmd.output, skipping it" << std::endl;
+    } else {
+        ostringstream cmdoutstream;
+        cmdoutstream << cmdOut.rdbuf();
+        response.set_response(cmdoutstream.str());
+        cout << "Command output is: " << std::endl;
+        cout << cmdoutstream.str() << std::endl;
+    }
+    cmdOut.close();
+    for (int i=0; i < request.expectedfiles_size(); i++) {
+        std::string expectedFile = request.expectedfiles(i);
+        ifstream output_file;
+        output_file.open(expectedFile, ios::in| ios::binary);
+        if (!output_file) {
+            cout << "Unexpected: expected file " << expectedFile
+                <<  " is not present!" << std::endl;
+        } else {
+            ostringstream expectedFileContent;
+            expectedFileContent << output_file.rdbuf();
+            eve_tools::File *outputFile = response.add_outputfiles();
+            outputFile->set_name(expectedFile);
+            outputFile->set_content(expectedFileContent.str());
+            output_file.close();
+        }
+    }
+    size = response.ByteSize() + CODED_STRM_HDR_LEN;
+    char *resp_buffer = new char [size];
+    google::protobuf::io::ArrayOutputStream aos(resp_buffer, size);
+    CodedOutputStream *coded_output = new CodedOutputStream(&aos);
+    coded_output->WriteVarint32(response.ByteSize());
+    response.SerializeToCodedStream(coded_output);
+    send(sock, (void *)resp_buffer, size, 0);
+}
+
+int main(int argc, char *argv[])
+{
+
+    //Listen at IN_ADDR_ANY, SERVER_PORT
+    struct sockaddr_in serverAddr;
+    memset(&serverAddr, 0, sizeof(serverAddr));
+    serverAddr.sin_family = AF_INET;
+    serverAddr.sin_port = htons(SERVER_PORT);
+    serverAddr.sin_addr.s_addr = htonl(INADDR_ANY);
+
+    int listenSock;
+    if ((listenSock = socket(PF_INET, SOCK_STREAM, 0)) < 0) {
+        cerr << "could not create listen socket" << std::endl;
+        return -1;
+    }
+
+    if ((bind(listenSock, (struct sockaddr *)&serverAddr,
+                    sizeof(serverAddr))) < 0) {
+        cerr << "Bind failure, aborting" << std::endl;
+        return -1;
+    }
+
+    if (listen(listenSock, LISTEN_BACKLOG_LIMIT) < 0) {
+        cerr << "Listen failure, aborting" << std::endl;
+        return -1;
+    }
+
+    // socket address used to store client address
+    struct sockaddr_in clientAddr;
+    socklen_t clientAddrLen = 0;
+
+    while (true) {
+        // open a new socket to transmit data per connection
+        // Wait for an incoming connection.
+        // TBD: For now it is blocking call. Change it to non-blocking
```

Comment:
Don't you need to handle a connection from each domU as well?

Also, how would you identify the domU in the server?